zeroex/orderwatch/event_decoder.go

```diff
@@ -194,46 +198,82 @@ func NewDecoder() (*Decoder, error) {
 // from this contract address, the decoder will properly decode the `Transfer` and `Approve` events
 // including the correct event parameter names.
 func (d *Decoder) AddKnownERC20(address common.Address) {
+	d.knownERC20AddressesMu.Lock()
+	defer d.knownERC20AddressesMu.Unlock()
 	d.knownERC20Addresses[address] = true
 }
 
 // RemoveKnownERC20 removes an ERC20 address from the list of known addresses. We will no longer decode
 // events for this token.
 func (d *Decoder) RemoveKnownERC20(address common.Address) {
+	d.knownERC20AddressesMu.Lock()
+	defer d.knownERC20AddressesMu.Unlock()
 	delete(d.knownERC20Addresses, address)
 }
 
+// isKnownERC20 checks if the supplied address is a known ERC20 contract
+func (d *Decoder) isKnownERC20(address common.Address) bool {
+	d.knownERC20AddressesMu.Lock()
```

Comment:
`isKnownERC20` and other similar methods should use `RLock` and `RUnlock` since they are reading from the map but not writing to it.